File: kinoafisha/Legacy/Model/Cinema.m
//
//  Cinema.m
//  kinoafisha
//
//  Created by Eugene Dorfman on 11/9/14.
//  Copyright (c) 2014 justadreamer. All rights reserved.
//

#import "Cinema.h"

@implementation Cinema
+ (NSDictionary *)JSONKeyPathsByPropertyKey {
    return @{
        @"name":@"name",
        @"detailURL":@"link",
        @"thumbnailURL":@"thumbnail",
        @"address":@"address",
        @"phone":@"phone",
        @"rating":@"rating",
        @"votesCount":@"votes_count"
    };
}

+ (NSValueTransformer *)detailURLJSONTransformer {
    return [NSValueTransformer valueTransformerForName:MTLURLValueTransformerName];
}

+ (NSValueTransformer *)thumbnailURLJSONTransformer {
    return [NSValueTransformer valueTransformerForName:MTLURLValueTransformerName];
}

@end

File: kinoafisha/Cinema.swift
//
//  Cinema.swift
//  kinoafisha
//
//  Created by Eugene Dorfman on 10/7/19.
//  Copyright © 2019 justadreamer. All rights reserved.
//

import Foundation

struct Cinema: Codable {
    enum CodingKeys: String, CodingKey {
        case name
        case detailURL = "link"
        case thumbnailURL = "thumbnail"
        case address = "address"
        case phone = "phone"
        case rating = "rating"
        case votesCount = "votes_count"
    }

    var name: String
    var detailURL: URL
    var thumbnailURL: URL
    var address: String
    var phone: String
    var rating: String
    var votesCount: String
}
